app/views/deals/show.html.erb

```diff
@@ -29,9 +32,11 @@
         <br>
 
         <% if @deal.is_live? %>
-          <%= link_to I18n.t(:add_to_cart, scope: :action), deal_url(@deal), class: "btn btn-success btn-lg view-deal-btn" %>
+          <%= button_to I18n.t(:add_to_cart, scope: :action), line_items_url(deal_id: @deal), class: "btn btn-success btn-lg add-to-cart-btn", method: :post, data: { name: 'add-to-cart-btn' } %>
+          <p><small class="form-text text-muted">Can be ordered only once.</small></p>
+
         <% elsif @deal.is_expired? %>
```

Comment:
don't show add to cart button in case if a deal is expired. 
Show some appropriate message instead.